libuuu/buffer.cpp

```diff
@@ -142,55 +146,86 @@ class FSBasic
 	virtual std::shared_ptr<FragmentBlock> ScanCompressblock(const string& /*backfile*/, size_t& /*input_offset*/, size_t& /*output_offset*/) { return NULL; };
 	virtual int PreloadWorkThread(shared_ptr<FileBuffer>outp);
 
-	virtual int split(const string &filename, string *outbackfile, string *outfilename, bool dir=false)
-	{
-		string path = str_to_upper(filename);
+	/**
+	 * @brief Splits a file system path into directory and final name parts
+	 * @param path Input path
+	 * @param[out] dir_part Directory path info from input path
+	 * @param[out] name_part Final name part from input path
+	 * @param dir [what does this mean/imply?]
+	 * @return 0 for success; -1 for error
+	 * @details
+	 * If no/blank m_ext:
+	 *	 If dir:
+	 *     Load dir_part with directory path part of path; >./ if none
+	 *     Load name_part with file name part of path; blank if none
+	 *   Else:
+	 *     Load dir_part with input path
+	 * Else:
+	 *   ext = m_ext
+	 *   Append slash to ext if !dir
+	 *   Find last ext in path
+	 *   If not found: fail
+	 *   Load dir_part with path up to and including ext
+	 *   Load name_part with path after found ext
+	 * @example
+	 * path:"f",     m_ext:"",  !dir ==> ">./", "f"
+	 * path:"d/f",   m_ext:"",  !dir ==> "d", "f"
+	 * path:"d/",    m_ext:"",  !dir ==> "d", ""
+	 * path:"f",     m_ext:"z", !dir ==> error
+	 * path:"d/z/f", m_ext:"z", !dir ==> "d/z", "f"
+	 * path:"d/z/",  m_ext:"z", !dir ==> "d/z", ""
+	 * @note
+	 * Passing a bool like dir is bad style since it only controls flow.
+	 */
+	virtual int split(const string &path, string *dir_part, string *name_part, bool dir=false)
+	{
```

Comment:
use seperate patch to change outbackfile-> dir_part, outfilename -> filename_part